flake.nix

```diff
@@ -0,0 +1,187 @@
+{
+  description = "Build a cargo project";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    crane.url = "github:ipetkov/crane";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.rust-analyzer-src.follows = "";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    advisory-db = {
+      url = "github:rustsec/advisory-db";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    crane,
+    fenix,
+    flake-utils,
+    advisory-db,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+
+      inherit (pkgs) lib;
+      toolchain = fenix.packages.${system}.fromToolchainFile {
+        file = ./rust-toolchain.toml;
+        sha256 = "sha256-3jVIIf5XPnUU1CRaTyAiO0XHVbJl12MSx3eucTXCjtE=";
```

Comment:
out of date with main, needs to be `sha256-6ujCCaIqt4N0wj98YislhhLX3yB8eh2TAl671GJ8bR4=`

Reply:
I'll fix this right away.
We _might_ be able to automate stuff like this for the future. If nothing else, it shouldn't at all be difficult to create a CI to inform me when `Cargo.toml` or `rust-toolchain.toml` change so I can update the flake accordingly.

Reply:
Are you sure that this hash is out of date? From what I'm seeing, the current Main's `rust-toolchain.toml` produces the same hash as what's currently in the flake.

Reply:
when testing, i had to change the hash of the toolchain config to `sha256-yMuSb5eQPO/bHv+Bcf/US8LVMbf/G/0MSfiPwBhiPpk=` after the change from [pinning 1.80](https://github.com/justDeeevin/scope/blob/2993f1e1a820492285eab0375de577d0777ed2ce/rust-toolchain.toml) to [following stable](https://github.com/scopeclient/scope/blob/7f970dc2d8ddb9d29bd1f3aa0b8e3d473f971efe/rust-toolchain.toml)

Reply:
I forgot that my local main branch had `justdeeevin/posting/main` as its upstream, so it wasn't receiving new commits. :facepalm:

updating now